simapp/state.go

```diff
@@ -21,7 +21,7 @@ import (
 // AppStateFn returns the initial application state using a genesis or the simulation parameters.
 // It panics if the user provides files for both of them.
 // If a file is not given for the genesis or the sim params, it creates a randomized one.
-func AppStateFn(cdc codec.JSONMarshaler, simManager *module.SimulationManager) simtypes.AppStateFn {
+func AppStateFn(cdc codec.JSONMarshaler, legacyAmino *codec.LegacyAmino, simManager *module.SimulationManager) simtypes.AppStateFn {
```

Comment:
I think we actually don't want simulations to use LegacyAmino at all. I would revert this and refactor every place that amino is used in simulations